src/foam/core/AbstractArrayPropertyInfo.java

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license
+ * Copyright 2017 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package foam.core;
+
+import foam.nanos.logger.NanoLogger;
+import java.lang.UnsupportedOperationException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public abstract class AbstractArrayPropertyInfo
```

Comment:
This is used to handle FObjectArray and Array, which will be handled completely differently (simple values vs. objects). Since this doesn't work with primitive types anyway, I would just divide it into two different classes (or three, if you count Map). Also, how is Map being supported?